arrows/ocv/refine_detections_draw.cxx

```diff
@@ -0,0 +1,183 @@
+
+#include <kwiversys/SystemTools.hxx>
+
+#include <arrows/ocv/image_container.h>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+using namespace kwiver::vital;
+
+namespace kwiver {
+namespace arrows {
+namespace ocv {
+
+typedef kwiversys::SystemTools ST;
+
+/// Private implementation class
+class refine_detections_draw::priv
+{
+public:
+
+  /// Constructor
+  priv()
+  : pattern( "feature_tracks_%05d.png" ),
+    id( 0 )
+  {
+  }
+
+  /// Destructor
+  ~priv()
+  {
+  }
+
+  /// Parameters
+  std::string pattern;
+
+  /// Variables
+  unsigned id;
+};
+
+
+/// Constructor
+refine_detections_draw
+::refine_detections_draw()
+: d_( new priv() )
+{
+}
+
+
+/// Destructor
+refine_detections_draw
+::~refine_detections_draw()
+{
+}
+
+
+/// Get this algorithm's \link vital::config_block configuration block \endlink
+vital::config_block_sptr
+refine_detections_draw
+::get_configuration() const
+{
+  vital::config_block_sptr config = vital::algo::refine_detections::get_configuration();
+
+  config->set_value( "pattern", d_->pattern,
+                     "The output pattern for writing images to disk." );
+
+  return config;
+}
+
+
+/// Set this algorithm's properties via a config block
+void
+refine_detections_draw
+::set_configuration( vital::config_block_sptr in_config )
+{
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  d_->pattern = config->get_value<std::string>( "pattern" );
+}
+
+/// Check that the algorithm's currently configuration is valid
+bool
+refine_detections_draw
+::check_configuration(vital::config_block_sptr config) const
+{
+  return true;
+}
+
+/// Output images with tracked features drawn on them
+vital::detected_object_set_sptr
+refine_detections_draw
+::refine( vital::image_container_sptr image_data,
+          vital::detected_object_set_sptr detections ) const
+{
+  cv::Mat img = ocv::image_container::vital_to_ocv( image_data->get_image() );
+
+  VITAL_FOREACH( auto det, detections->select() )
+  {
+    // Generate output filename
+    std::string ofn;
+    size_t max_len = d_->pattern.size() + 4096;
+    ofn.resize( max_len );
+    int num_bytes = snprintf( &ofn[0], max_len, d_->pattern.c_str(), d_->id++ );
+
+    if( num_bytes < 0 )
+    {
+      LOG_WARN( logger(), "Could not format output file name: \"" <<
+        d_->pattern << "\". Disabling writing to disk." );
+    }
+
+    // Output image to file
+    vital::bounding_box_d bbox = det->bounding_box();
+
+    cv::Size s = img.size();
+    vital::bounding_box_d bounds( vital::bounding_box_d::vector_type( 0, 0 ),
+      vital::bounding_box_d::vector_type( s.width, s.height ) );
+
+    bbox = intersection( bounds, bbox );
+
+    // Make CV rect for out bbox coordinates
+    cv::Rect r( bbox.upper_left()[0], bbox.upper_left()[1],
+      bbox.width(), bbox.height() );
+
+    cv::Mat crop = img( r );
+    cv::imwrite( ofn, img );
```

Comment:
Class name implies that detections are drawn. It looks like they are clipped and written to disk. Should the class/file name be more descriptive?